controller/controller.go

```diff
@@ -584,84 +582,79 @@ func UpdateSchedulerConfig(
 
 	l.Info("starting scheduler update")
 
-	if maxSurge <= 0 {
-		return errors.New("invalid parameter: maxsurge must be greater than 0")
+	err := validateConfig(logger, configYAML, maxSurge)
+	if err != nil {
+		return err
 	}
 
-	if configYAML.AutoScaling.Max > 0 && configYAML.AutoScaling.Min > configYAML.AutoScaling.Max {
-		return errors.New("invalid parameter: autoscaling max must be greater than min")
-	}
+	// Lock watchers so they don't scale up or down and the scheduler is not
+	// overwritten with older version on database
+	// during the databse write phase
+	globalLockKey := models.GetSchedulerLockKey(config.GetString("watcher.lockKey"), configYAML.Name)
+	globalLock, canceled, err := acquireLock(
+		ctx,
+		logger,
+		clock,
+		redisClient,
+		config,
+		operationManager,
+		globalLockKey,
+		schedulerName,
+	)
+
+	defer releaseLock(
+		logger,
+		redisClient,
+		globalLock,
+		schedulerName,
+	)
 
-	// if using resource scaling (cpu, mem) requests must be set
-	err := validateMetricsTrigger(configYAML, logger)
 	if err != nil {
 		return err
 	}
 
-	// Lock watchers so they don't scale up or down and the scheduler is not
-	//  overwritten with older version on database
-	var lock *redisLock.Lock
-	timeoutDur := time.Duration(timeoutSec) * time.Second
-	willTimeoutAt := clock.Now().Add(timeoutDur)
-	ticker := time.NewTicker(2 * time.Second)
-	defer ticker.Stop()
-	timeout := time.NewTimer(timeoutDur)
-	defer timeout.Stop()
+	if canceled {
+		return nil
+	}
 
-waitForLock:
-	for {
-		lock, err = redisClient.EnterCriticalSection(
-			redisClient.Trace(ctx),
-			lockKey,
-			time.Duration(lockTimeoutMS)*time.Millisecond,
-			0, 0,
-		)
-		select {
-		case <-timeout.C:
-			return errors.New("timeout while wating for redis lock")
-		case <-ticker.C:
-			if operationManager.WasCanceled() {
-				l.Warn("operation was canceled")
-				return nil
-			}
+	// Lock updates on scheduler during all the process
+	configLockKey := models.GetConfigLockKey(config.GetString("watcher.lockKey"), configYAML.Name)
```

Comment:
The same about naming goes here. `models` is not exclusive to schedulers. What do you think about `GetSchedulerConfigLockKey` ?.